Source/NSPathControl.m

```diff
@@ -0,0 +1,565 @@
+      NSURL *url = [urls objectAtIndex: 0];
+      [self setURL: url];
+    }
+
+  [[sender menu] close];
+  // AUTORELEASE([sender menu]);
+}
+
+- (void) mouseDown: (NSEvent *)event
+{
+  if (![self isEnabled])
+    {
+      [super mouseDown: event];
+      return;
+    }
+
+  if ([self pathStyle] == NSPathStylePopUp)
+    {
+      NSPathCell *acell = (NSPathCell *)[self cell];
+      NSArray *array = [acell pathComponentCells];
+      NSMenu *menu = AUTORELEASE([[NSMenu alloc] initWithTitle: nil]);
+      NSPathComponentCell *c = nil;
+      NSEnumerator *en = [array objectEnumerator];
+      
+      while((c = [en nextObject]) != nil)
+        {
+          NSURL *u = [c URL];
+          NSString *s = [[u path] lastPathComponent];
+          NSMenuItem *i = [[NSMenuItem alloc] init];
+
+          [i setTitle: s];
+          [i setTarget: self];
+          [i setAction: @selector(_doMenuAction:)];
+          [i setRepresentedObject: c];
+          
+          [menu insertItem: i
+                   atIndex: 0]; 
+          RELEASE(i);
+        }
+
+      // Add separator
+      [menu insertItem: [NSMenuItem separatorItem]
+               atIndex: 0];
+
+      // Add choose menu option
+      NSMenuItem *i = [[NSMenuItem alloc] init];
+      [i setTitle: @"Choose..."];
+      [i setTarget: self];
+      [i setAction: @selector(_doChooseMenuAction:)];
+      [menu insertItem: i
+               atIndex: 0];
+      RELEASE(i);
+
+      [self setMenu: menu];
+      // RELEASE(menu);
+      
+      if (_delegate)
+        {
+          if ([(id)_delegate respondsToSelector: @selector(pathControl:willPopUpMenu:)])
+            {
+              [_delegate pathControl: self
+                       willPopUpMenu: menu];
+            }
+        }
+      
+      if ([_cell delegate])
+        {
+          if ([(id)[_cell delegate] respondsToSelector: @selector(pathCell:willPopUpMenu:)])
+            {
+              [[_cell delegate] pathCell: _cell
+                           willPopUpMenu: menu];
+            }
+        }
+
+      
+      [menu popUpMenuPositionItem: [menu itemAtIndex: 0]
+                       atLocation: NSMakePoint(0.0, 0.0)
+                           inView: self];
+    }
+  else
+    {
+      NSPathComponentCell *pcc = [_cell pathComponentCellAtPoint: [event locationInWindow]
+                                                       withFrame: [self frame]
+                                                          inView: self];
+      [_cell _setClickedPathComponentCell: pcc];
+      if ([[self cell] action])
+        {
+          [self sendAction: [[self cell] action]
+                        to: [[self cell] target]];
+        }
+    }
+}
+
+- (NSDragOperation) draggingEntered: (id<NSDraggingInfo>)sender
+{
+  if (_delegate != nil)
+    {
+      NSDragOperation dop = [_delegate pathControl: self
+                                      validateDrop: sender];
+      return dop;
+    }
+  
+  return NSDragOperationNone;
```

Comment:
Why none? Even without a delegate dragging should be possible.

Reply:
I just figured that might be the default..

Reply:
Changed to Copy until I can test it more.